sycl/source/handler.cpp

```diff
@@ -1412,6 +1412,31 @@ checkContextSupports(const std::shared_ptr<detail::context_impl> &ContextImpl,
   return SupportsOp;
 }
 
+void handler::verifyDeviceHasProgressGuarantee(
+    sycl::ext::oneapi::experimental::forward_progress_guarantee Guarantee,
+    sycl::ext::oneapi::experimental::execution_scope threadScope,
+    sycl::ext::oneapi::experimental::execution_scope CoordinationScope) const {
+  using execution_scope = sycl::ext::oneapi::experimental::execution_scope;
+  if (!MQueue->getDeviceImplPtr()->supportsForwardProgress(
+          Guarantee, threadScope, CoordinationScope)) {
+    if (threadScope == execution_scope::work_group) {
+      throw sycl::exception(
+          sycl::errc::feature_not_supported,
+          "Required progress guarantee for work groups is not "
+          "supported by your device!");
```

Comment:
I'm not sure what our convention is here, but I think "this device" looks better than "your device!".  Same comment for the other two exceptions.

```suggestion
          "supported by this device.");
```